Library/Homebrew/cask/lib/hbc/cask.rb

```diff
@@ -85,6 +85,16 @@ def installed?
       !versions.empty?
     end
 
+    def latest_installed_version
+      versions.last
+    end
+
+    def outdated?(greedy)
+      (greedy || !auto_updates && !version == :latest) &&
+      !version.nil? &&  # ignore Casks without sources
```

Comment:
Is there much difference semantically to instead swapping the two lines? Happy to put this first, but I don't understand what the `return false if` adds.

If anything, perhaps pulling this out to a small method e.g. `has_source` would be better from a self documenting perspective.

Happy to go whichever way on this as it seems mostly preferential.